pkg/api/resource.go

```diff
@@ -326,6 +336,99 @@ func (h APIHandler) ResourceStatusCheck(c *gin.Context) {
 
 }
 
+func (a ByCreatedAt) Len() int           { return len(a) }
+func (a ByCreatedAt) Swap(i, j int)      { a[i], a[j] = a[j], a[i] }
+func (a ByCreatedAt) Less(i, j int) bool { return a[i].CreatedAt.Before(a[j].CreatedAt) }
+
+func (h APIHandler) LastTaskLog(c *gin.Context) {
+	clusterName := c.Param("cluster_name")
+	clusterID := h.getClusterID(clusterName)
+
+	if clusterID == 0 {
+		c.JSON(http.StatusUnprocessableEntity, response(http.StatusUnprocessableEntity, fmt.Sprintf("cluster_name '%s' not found", clusterName), nil))
+		return
+	}
+
+	resourceName := c.Param("name")
+	namespace := c.Param("namespace")
+
+	config, err := getVclusterConfig(h.clientset, "internal", clusterName)
+	if err != nil {
+		c.JSON(http.StatusUnprocessableEntity, response(http.StatusUnprocessableEntity, err.Error(), nil))
+		return
+	}
+
+	clientset := kubernetes.NewForConfigOrDie(config)
+
+	// get the pods with a certain label in the default namespace
+	labelSelector := "terraforms.tf.galleybytes.com/resourceName=" + resourceName // change this to your label selector
+	pods, err := clientset.CoreV1().Pods(namespace).List(context.Background(), metav1.ListOptions{LabelSelector: labelSelector})
+	if err != nil {
+		log.Fatal(err)
```

Comment:
This will fail the API server. Instead, an http response should be sent mentioning what went wrong.